src/modules/EmailRecoveryModule.sol

```diff
@@ -0,0 +1,175 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.25;
+
+import { ERC7579ExecutorBase } from "@rhinestone/modulekit/src/Modules.sol";
+import { IERC7579Account } from "erc7579/interfaces/IERC7579Account.sol";
+import { IModule } from "erc7579/interfaces/IERC7579Module.sol";
+import { ExecutionLib } from "erc7579/lib/ExecutionLib.sol";
+import { ModeLib } from "erc7579/lib/ModeLib.sol";
+
+import { IRecoveryModule } from "../interfaces/IRecoveryModule.sol";
+import { IEmailRecoveryManager } from "../interfaces/IEmailRecoveryManager.sol";
+import { ISafe } from "../interfaces/ISafe.sol";
+import { BytesLib } from "../libraries/BytesLib.sol";
+
+contract EmailRecoveryModule is ERC7579ExecutorBase, IRecoveryModule {
+    using BytesLib for bytes;
+    /*//////////////////////////////////////////////////////////////////////////
+                                    CONSTANTS
+    //////////////////////////////////////////////////////////////////////////*/
+
+    address public immutable emailRecoveryManager;
+
+    event NewValidatorRecovery(address indexed validatorModule, bytes4 recoverySelector);
+
+    error NotTrustedRecoveryManager();
+    error InvalidSubjectParams();
+    error InvalidValidator(address validator);
+    error InvalidSelector(bytes4 selector);
+
+    mapping(address validatorModule => mapping(address account => bytes4 allowedSelector)) internal
+        allowedSelectors;
+    mapping(address account => address validator) internal validators;
```

Comment:
I assume this is just for PoC and in the final version we will support multiple validators

Reply:
Yes - just pushed an initial commit for this. It's a little messy with multiple mappings but wanted to push something as a POC.

Let me know if you can think of any cleaner ways of doing this